src/graphnet/training/callbacks.py

```diff
@@ -152,3 +158,92 @@ def on_train_epoch_end(
             h.setLevel(logging.ERROR)
             logger.info(str(super().train_progress_bar))
             h.setLevel(level)
+
+
+class GraphnetEarlyStopping(EarlyStopping):
+    """Early stopping callback for graphnet."""
+
+    def __init__(self, save_dir: str, **kwargs: Dict[str, Any]) -> None:
+        """Construct `GraphnetEarlyStopping` Callback.
+
+        Args:
+            save_dir: Path to directory to save best model and config.
+            **kwargs: Keyword arguments to pass to `EarlyStopping`. See
+                `pytorch_lightning.callbacks.EarlyStopping` for details.
+        """
+        self.save_dir = save_dir
```

Comment:
Do we need to specify the `save_dir `here? From [the documentation](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.callbacks.ModelCheckpoint.htm) of `ModelCallback` it says it will automatically default to the directory of the Trainer.

Reply:
I could probably change the callback to get the directory from the trainer, or maybe have that be a fallback option. 

This callback is only subclassing EarlyStopping, and adding functionality such that whenever the "best_score" is changed it will save the statedict to disk, and it also saves the model_config to disk at the start of the run.

The idea behind this behavior over ModelCheckpoint, is to be more inline, with how models are best saved withing GraphNeT. ModelCheckpoints store more data, and are not straight forward to load in, when the hyperparameters of a lightning model is saved with `save_hyperparameters()`.

ModelCheckpoints to my understanding are more-so used to be able to "interrupt" a training and resume it at a later point, which is usefull when running on clusters where you might reserve a node for less time, than what it takes to finish a training.